### DISCUSSION TITLE
defining a mission to be offered if another one isnt

### DISCUSSION THREAD

**Discussion by daggs1** [endless-sky/endless-sky#9860] [General]
Greetings,

I'm working on a small storyline where I want to add a mission that should be offered if mission a wasn't offered and only 50% of the time.
so I added this in the mission definition:
 	to offer
 		not "mission a: offered"
 		random < 50 

saved, booted the games and entered and existing several jobs boards in several systems multiple time, mission a wasn't offered at any of the boards (it can be offered) and my mission wasn't offered too, I'm assuming that I did something wrong in the definition.
I haven't see any mission error prints on the loading part so I assume the syntax is ok

help is appreciated,

Thanks

**Comment by daggs1** [comment]
ok, I've reread my question and understood that what I've asked and what I wanted to ask are not the same, so I'll rephrase:
when the player opens the jobs board, he sees a list of jobs, lets say there is a "mission A" that appears or not in the jobs board.
my question is, can I define a job "mission b" that will appear in the jobs board 50% of the times "mission A" is not shown in the jobs board?

**Comment by daggs1** [comment]
so after some tinkering, looks like what I need is this:
to offer
    not "mission a: available"
    random < 25

**Comment by daggs1** [comment]
:( I stand corrected, I just got Mission A and Mission B in the same time

**Comment by mOctave** [comment]
I'm not sure there's a good way to do this. However, if you only have two missions, you could set them up so that they only offer on alternate days, and one of them has a 50% chance of offering. Most players probably wouldn't notice the pattern in the missions being offered.

**Comment by daggs1** [comment]
> I'm not sure there's a good way to do this. However, if you only have two missions, you could set them up so that they only offer on alternate days, and one of them has a 50% chance of offering. Most players probably wouldn't notice the pattern in the missions being offered.

one is a standard mission, I'd prefer not to change it, btw, how can I set that in a mission?

**Reply by mOctave** [reply]
I think
```
mission "Mission A"
	to offer
		"day" / 2 * 2 == "day"

mission "Mission B"
	to offer
		not
			"day" / 2 * 2 == "day"
```
should do it, but I haven't tested.
